cub/cub/device/device_find_if.cuh

```diff
@@ -83,14 +93,35 @@ __global__ void find_if(IterBegin begin, IterEnd end, Pred pred, int* result, st
       return;
     }
 
+    //// vectorized loads begin
+    InputT* d_in_unqualified = const_cast<InputT*>(begin) + tile_offset + (threadIdx.x * _VECTOR_LOAD_LENGTH);
+
+    cub::CacheModifiedInputIterator<cub::CacheLoadModifier::LOAD_DEFAULT, VectorT> d_vec_in(
+      reinterpret_cast<VectorT*>(d_in_unqualified));
+
+    InputT input_items[elements_per_thread];
+    VectorT* vec_items = reinterpret_cast<VectorT*>(input_items);
+
+#pragma unroll
+    for (int i = 0; i < WORDS; ++i)
+    {
+      vec_items[i] = d_vec_in[block_threads * i];
+    }
+    //// vectorized loads end
+    // what about when input is not devisible by _VECTOR_LOAD_LENGTH? --> case for not full tile to be added
+
     bool found = false;
     for (int i = 0; i < elements_per_thread; ++i)
     {
-      auto index = tile_offset + threadIdx.x + i * blockDim.x;
+      int index = i % WORDS + (i / WORDS) * block_threads * WORDS + threadIdx.x * WORDS + tile_offset;
```

Comment:
The expression for the evaluation of this index can of course be simplified further. It is as is for educational purposes.